Application/Api/Education/ClassRegister/ApiDiary.php

```diff
@@ -0,0 +1,585 @@
+        }
+
+        if (Diary::useService()->updateDiary($tblDiary, $Data)) {
+            return new Success('Der Eintrag wurde erfolgreich gespeichert.')
+                . self::pipelineLoadDiaryContent($tblDivision ? $tblDivision->getId() : null, $tblGroup ? $tblGroup->getId() : null)
+                . self::pipelineClose();
+        } else {
+            return new Danger('Der Eintrag konnte nicht gespeichert werden.') . self::pipelineClose();
+        }
+    }
+
+    /**
+     * @param $DiaryId
+     *
+     * @return string
+     */
+    public function openDeleteDiaryModal($DiaryId)
+    {
+        if (!($tblDiary = Diary::useService()->getDiaryById($DiaryId))) {
+            return new Danger('Der Eintrag wurde nicht gefunden', new Exclamation());
+        }
+
+        return new Title(new Remove() . ' Eintrag löschen')
+            . new Layout(
+                new LayoutGroup(
+                    new LayoutRow(
+                        new LayoutColumn(
+                            new Panel(
+                                new Question() . ' Diesen Eintrag wirklich löschen?',
+                                array(
+                                    $tblDiary->getDate(),
+                                    $tblDiary->getSubject(),
+                                    $tblDiary->getContent()
+                                ),
+                                Panel::PANEL_TYPE_DANGER
+                            )
+                            . (new DangerLink('Ja', self::getEndpoint(), new Ok()))
+                                ->ajaxPipelineOnClick(self::pipelineDeleteDiarySave($DiaryId))
+                            . (new Standard('Nein', self::getEndpoint(), new Remove()))
+                                ->ajaxPipelineOnClick(self::pipelineClose())
+                        )
+                    )
+                )
+            );
+    }
+
+    /**
+     * @param $DiaryId
+     *
+     * @return Danger|string
+     */
+    public function saveDeleteDiaryModal($DiaryId)
+    {
+        if (!($tblDiary = Diary::useService()->getDiaryById($DiaryId))) {
+            return new Danger('Der Eintrag wurde nicht gefunden', new Exclamation());
+        }
+        $tblDivision = $tblDiary->getServiceTblDivision();
+        $tblGroup = $tblDiary->getServiceTblGroup();
+
+        if (Diary::useService()->destroyDiary($tblDiary)) {
+            return new Success('Der Eintrag wurde erfolgreich gelöscht.')
+                . self::pipelineLoadDiaryContent($tblDivision ? $tblDivision->getId() : null, $tblGroup ? $tblGroup->getId() : null)
+                . self::pipelineClose();
+        } else {
+            return new Danger('Der Eintrag konnte nicht gelöscht werden.') . self::pipelineClose();
+        }
+    }
+
+    /**
+     * @param int|null $DivisionId
+     * @param int|null $GroupId
+     *
+     * @return Danger|string
+     */
+    public function openSelectStudentModal($DivisionId = null, $GroupId = null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Incorrect spacing between argument "$DivisionId" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5646354)